misc/prometheus/mzcompose.py

```diff
@@ -0,0 +1,26 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+from materialize.mzcompose.services import Service
+
+SERVICES = [
+    Service(
+        "prometheus",
+        {
+            "image": "prom/prometheus:v2.41.0",
+            "ports": ["9090:9090"],
+            "volumes": [
+                "./prometheus.yml:/etc/prometheus/prometheus.yml",
+                "../../mzdata/prometheus:/mnt/services",
```

Comment:
I am afraid there is a permission issue of some sort. The mzdata directory is created, owned by `root`

```
pstoev@Ubuntu-2004-focal-64-minimal:~/benesch-process-monitoring$ ls -la mzdata/prometheus/
total 8
drwxr-xr-x 2 root root 4096 Dec 27 10:34 .
drwxr-xr-x 3 root root 4096 Dec 27 10:34 ..
```

And therefore bin/environmentd is unable to start:
```
Traceback (most recent call last):
  File "/home/pstoev/benesch-process-monitoring/misc/python/materialize/cli/run.py", line 137, in main
    environment_id = environment_file.read_text().rstrip()
  File "/usr/lib/python3.9/pathlib.py", line 1266, in read_text
    with self.open(mode='r', encoding=encoding, errors=errors) as f:
  File "/usr/lib/python3.9/pathlib.py", line 1252, in open
    return io.open(self, mode, buffering, encoding, errors, newline,
  File "/usr/lib/python3.9/pathlib.py", line 1120, in _opener
    return self._accessor.open(self, flags, mode)
FileNotFoundError: [Errno 2] No such file or directory: '/home/pstoev/benesch-process-monitoring/mzdata/environment-id'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/pstoev/benesch-process-monitoring/misc/python/materialize/cli/run.py", line 239, in <module>
    main()
  File "/home/pstoev/benesch-process-monitoring/misc/python/materialize/cli/run.py", line 140, in main
    environment_file.write_text(environment_id)
  File "/usr/lib/python3.9/pathlib.py", line 1285, in write_text
    with self.open(mode='w', encoding=encoding, errors=errors) as f:
  File "/usr/lib/python3.9/pathlib.py", line 1252, in open
    return io.open(self, mode, buffering, encoding, errors, newline,
  File "/usr/lib/python3.9/pathlib.py", line 1120, in _opener
    return self._accessor.open(self, flags, mode)
PermissionError: [Errno 13] Permission denied: '/home/pstoev/benesch-process-monitoring/mzdata/environment-id'
```

Reply:
Ah, yeah, you have to start them in the opposite order: first `bin/environmentd`, and _then_ `./mzcompose run default`. I think we can fix that by creating the directory before the Docker daemon though, in mzcompose. Sec.

Reply:
Fixed!